mapproxy/source/ogcapitiles.py

```diff
@@ -0,0 +1,367 @@
+        self,
+        source_name,
+        configuration_context,
+        landingpage_url,
+        collection,
+        http_client,
+        tile_matrix_set_id=None,
+        coverage=None,
+        image_opts=None,
+        error_handler=None,
+        res_range=None,
+    ):
+        MapLayer.__init__(self, image_opts=image_opts)
+        self.source_name = source_name
+        self.configuration_context = copy.copy(configuration_context)
+        self.configuration_context.grids = {}
+        self.configuration_context.sources = {}
+        self.configuration_context.caches = {}
+        self.landingpage_url = landingpage_url.rstrip("/")
+        self.collection = collection
+        self.http_client = http_client
+        self.image_opts = image_opts or ImageOptions()
+        self.tile_matrix_set_id = tile_matrix_set_id
+        self.coverage = coverage
+        if self.coverage:
+            self.extent = MapExtent(self.coverage.bbox, self.coverage.srs)
+        else:
+            self.extent = DefaultMapExtent()
+        self.res_range = res_range
+        self.error_handler = error_handler
+
+        self.lock = Lock()
+        # Below variables are protected under the lock
+        self._reset_caches()
+
+    def _reset_caches(self):
+        self.has_got_tileset_list = False
+        self.map_crs_to_tilesets_list = {}
+        self.map_srs_to_grid_and_template_url = {}
+        self.map_grid_name_to_cachemaplayer = {}
+
+    def _build_url(self, href):
+        return build_absolute_url(self.landingpage_url, href)
+
+    def _get_tileset_list(self):
+        with self.lock:
+            # Used by tests to avoid caching of "metadata"
+            global reset_cache
+            if reset_cache:
+                reset_cache = False
+                self._reset_caches()
+
+            if self.has_got_tileset_list:
+                return
+            self.has_got_tileset_list = True
+
+            headers = {"Accept": "application/json"}
+
+            url = self.landingpage_url
+            if self.collection:
+                url += "/collections/" + self.collection
+            try:
+                resp = self.http_client.open(url, headers=headers)
+            except HTTPClientError as e:
+                log.warning(f"Cannot retrieve {url}: %s", e)
+                reraise_exception(SourceError(e.args[0]), sys.exc_info())
+            try:
+                j = json.loads(resp.read().decode("utf-8"))
+            except Exception as e:
+                log.warning(f"Cannot parse response to {url} as JSON: %s", e)
+                reraise_exception(SourceError(e.args[0]), sys.exc_info())
+
+            if "links" not in j:
+                ex = SourceError(f"Could not retrieve 'links' in {url} response")
+                log.error(ex)
+                raise ex
+
+            tilesets_map_href = find_href_in_links(
+                j["links"],
+                "http://www.opengis.net/def/rel/ogc/1.0/tilesets-map",
```

Comment:
according to the spec, the value `[ogc-rel:tilesets-map]` should also be allowed / supported